accepted/0000-workspace-run-commands.md

```diff
@@ -0,0 +1,57 @@
+- Start Date: 2017-09-21
+- RFC PR:
+- Yarn Issue:
+
+# Summary
+
+Allow Yarn CLI to execute a package script on each packages from the workspace root.  
+
+# Motivation  
+
+"Lerna". Lerna does a great job for handling monorepos. Since Yarn has a built-in 
+workspaces feature, we could use some of the functionalities that being used in 
+lerna (not entirely though!) for handling the monorepo more efficiently.  
+
+Just like installing all dependencies from one place (workspace root), it'd be great 
+to execute all the sub-package scripts from the workspace root using a single command. 
+Not only this would make a less back-and-forth traveling between the packages to 
+execute scripts, but also it'll help a lot for CI/CD configuration. 
+
+
+# Detailed design  
+
+As @BYK mentioned [here](https://github.com/yarnpkg/yarn/issues/4467#issuecomment-330873337), 
+we could create two commands for this feature.  
+
+* `yarn workspaces list`  
+* `yarn workspaces run <command>`
+
+By introducing `workpaces` command, it brings a modular approach from the CLI's perspective.  
```

Comment:
Possible additions:

*yarn workspaces add <name>*: This would create the folder if it doesn't exist and make sure it is included in the top-level `workspaces` config.
*yarn workspaces remove <name>*: Remove a specific workspace from the config and optionally delete the folder if `-D` flag is passed.

Reply:
For the `yarn workspaces add`, shouldn't we pass the workspace name too?

Reply:
I believe yarn workspaces run <command> is already implemented via https://github.com/yarnpkg/rfcs/blob/master/implemented/0000-workspaces-command.md#yarn-workspace-workspace-name-command-name-?